src/deep_learning_container.py

```diff
@@ -135,14 +190,39 @@ def query_bucket():
     return response
 
 
-def requests_helper(url, timeout):
-    response = None
-    try:
-        response = requests.get(url, timeout=timeout)
-    except requests.exceptions.RequestException as e:
-        logging.error("Request exception: {}".format(e))
-
-    return response
+def tag_instance():
+    """
+    Apply instance tag on the instance that is running the container using botocore
+    """
+    instance_id = _retrieve_instance_id()
+    region = _retrieve_instance_region()
+    args = parse_args()
+    framework, framework_version, container_type = args.framework, args.framework_version, args.container_type
+    py_version = sys.version.split(" ")[0]
+    device = _retrieve_device()
+    cuda_version = f"_cuda{_retrieve_cuda()}" if device == "gpu" else ""
+    os_version = _retrieve_os()
+
+    tag = f"{framework}_{container_type}_{framework_version}_python{py_version}_{device}{cuda_version}_{os_version}"
+    tag_struct = {"Key": "aws-dlc-autogenerated-tag-do-not-delete", "Value": tag}
+
+    request_status = None
+    if instance_id and region:
+        try:
+            session = botocore.session.get_session()
+            ec2_client = session.create_client("ec2", region_name=region)
+            response = ec2_client.create_tags(Resources=[instance_id], Tags=[tag_struct])
+            request_status = response.get("ResponseMetadata").get("HTTPStatusCode")
```

Comment:
why not use `response.raise_for_status()`?

Reply:
In this case, `response` is a dict as per the convention in https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/ec2.html#EC2.Client.create_tags.

I return the request HTTP status code just to make the function signature equivalent for `tag_instance` and `query_bucket`.

Reply:
ahh I mistake this is wrriten in `requests`.  Would raising Exception cause anything happen?

Reply:
It would still be invisible to a user that runs this script indirectly through `import <framework>`, but not to those who directly execute this script, because the `import <framework>` statements have been written to suppress the output from the process that executes this script.

However, I'd like to force this script to run as quietly as possible regardless, so that it fails silently even if the parent process doesn't suppress its output.